process/block/preprocess/intermediateResults.go

```diff
@@ -237,6 +237,25 @@ func (irp *intermediateResultsProcessor) CreateMarshalizedData(txHashes [][]byte
 	return mrsTxs, nil
 }
 
+// GetAllCurrentUsedTxs returns the cached finalized transactions for current round
+func (irp *intermediateResultsProcessor) GetAllCurrentUsedTxs() map[string]data.TransactionHandler {
+	irp.mutInterResultsForBlock.Lock()
+
+	scrPool := make(map[string]data.TransactionHandler)
+	for txHash, txInfo := range irp.interResultsForBlock {
+		if txInfo.receiverShardID != irp.shardCoordinator.SelfId() {
+			continue
+		}
+		if txInfo.senderShardID != irp.shardCoordinator.SelfId() {
+			continue
+		}
+		scrPool[txHash] = txInfo.tx
```

Comment:
You add just the transactions that are intrashard? The ones that have both sender and receiver as SelfId?

Reply:
Might add some bool vars like:
```
isDestInOtherShard := txInfo.receiverShardID != irp.shardCoordinator.SelfId()
isSenderInOtherShard := txInfo.senderShardID != irp.shardCoordinator.SelfId()
```

Reply:
wanted to add only those which are finalized, otherwise, they will get added 2 times.

Reply:
if I add those bool's I have to change on the whole project.

Reply:
ok